lantiq: add support for Arcadyan VRV9510KWAC23

danielhuici opened the pull request:
The Arcadyan VRV9510KWAC23 (commercial name Livebox Next) is a Lantiq router distributed by Orange Spain

Hardware:
-   SoC: Lantiq VRX200
-   CPU: 2x MIPS 34Kc 500 MHz
-   RAM: 256 MiB DDR2
-   Flash: 128 MiB NAND
-   Ethernet: Built-in Gigabit Ethernet switch, 5x 1GbE
-   Wifi 2.4GHz: Broadcom BCM43222KFBG 802.11b/g/b MIMO 2T2R
-   Wifi 5GHz: Broadcom BCM4360KMLG 802.11ac MIMO 3T3R
-   USB: 2x USB 2.0
-   DSL: Built-in VDSL/ADSL2+ XWAY VRX208
-   LEDs: 8x
-   Buttons: 4x
-   Phone: Lantiq PEF 42068 V XWAY SLIC120

Install instructions:
Detailed instructions can be found on the wiki https://openwrt.org/toh/arcadyan/vrv9510kwac23
1. Boot into UART mode and upload the the https://raw.githubusercontent.com/danielhuici/arcadyan-vrv9510kwac23-utils/main/u-boot.asc file via serial console to boot into U-Boot.
2. Perform a backup of the NAND
3. Setup a TFTP server and serve the https://github.com/danielhuici/arcadyan-vrv9510kwac23-utils/raw/main/u-boot.ltq.lzo.nandspl. Replace the OEM bootloader with this one. Erase your NAND and write the image into it
4. Reboot the router
5. Serve the OpenWrt ramdisk image on your TFTP server and boot it via U-Boot
6. When OpenWrt boots, flash the SquashFS OpenWrt image using LuCi interface, so OpenWrt gets installed into the NAND

danielhuici commented:
There was an opened PR already. Accidentally I made something wrong with git and the bot made a merge of nothing (yeah git is my passion).
https://github.com/openwrt/openwrt/pull/11474

jospezial commented:
> There was an opened PR already. Accidentally I made something wrong with git and the bot made a merge of nothing (yeah git is my passion). #11474

Maybe it would have been better to repair the old PR.

danielhuici commented:
> > There was an opened PR already. Accidentally I made something wrong with git and the bot made a merge of nothing (yeah git is my passion). #11474
> 
> Maybe it would have been better to repair the old PR.

Afaik, there is no way to repair that, so when a PR is merged there is no way to open it back.

jospezial commented:
> * Wifi 5GHz: Broadcom BCM4360KMLG 802.11ac MIMO 3T3R

You forgot to edit that in your commit.

danielhuici commented:
> > * Wifi 5GHz: Broadcom BCM4360KMLG 802.11ac MIMO 3T3R
> 
> You forgot to edit that in your commit.
@jospezial You're right. It's fixed now

Noltari reviewed (approved):
LGTM

danielhuici commented:
New year's ping @hauke @Ansuel

danielhuici commented:
February's ping @hauke @Ansuel

hauke commented on target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vrv9510kwac23.dts:
```diff
@@ -0,0 +1,356 @@
+&vmmc {
+	status = "okay";
+	gpios = <&gpio 30 GPIO_ACTIVE_HIGH  //fxs relay
+		 &gpio 31 GPIO_ACTIVE_HIGH  //still unknown
```
What do you mean with `still unknown`?

danielhuici replied:
Those lines have been copied from Arcadyan VGV7519 (https://github.com/openwrt/openwrt/blob/master/target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vgv7519.dtsi)

Afaik, this router shares the same GPIO pins for vmmc @hauke

hauke commented on target/linux/lantiq/files/arch/mips/boot/dts/lantiq/vr9_arcadyan_vrv9510kwac23.dts:
```diff
@@ -0,0 +1,356 @@
+		};
+		internet_green: internet_green {
+			label = "green:internet";
+			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
```
Please replace the `label` attribute in the LED definition with the new definition:
For this LED it would look like this:
````
		internet_green: internet_green {
			gpios = <&gpio 26 GPIO_ACTIVE_LOW>;
			color = <LED_COLOR_ID_GREEN>;
			function = LED_FUNCTION_WAN;
		}
````

danielhuici replied:
Done

danielhuici committed:
lantiq: add support for Arcadyan VRV9510KWAC23

The Arcadyan VRV9510KWAC23 (trade name Livebox Next) is a Lantiq router distributed by some spanish ISPs

Hardware:
-   SoC: Lantiq VRX200
-   CPU: 2x MIPS 34Kc 500 MHz
-   RAM: 256 MiB DDR2
-   Flash: 128 MiB NAND
-   Ethernet: Built-in Gigabit Ethernet switch, 5x 1GbE
-   Wifi 2.4GHz: Broadcom BCM43222KFBG 802.11b/g/b MIMO 2T2R
-   Wifi 5GHz: Broadcom BCM4360KMLG 802.11ac MIMO 3T3R
-   USB: 2x USB 2.0
-   DSL: Built-in VDSL/ADSL2+ XWAY VRX208
-   LEDs: 8x
-   Buttons: 4x
-   Phone: Lantiq PEF 42068 V XWAY SLIC120

Install instructions:
Detailed instructions can be found on the wiki https://openwrt.org/toh/arcadyan/vrv9510kwac23
1. Boot into UART mode and upload the the https://raw.githubusercontent.com/danielhuici/arcadyan-vrv9510kwac23-utils/main/u-boot.asc file via serial console to boot into U-Boot.
2. Perform a backup of the NAND
3. Setup a TFTP server and serve the https://github.com/danielhuici/arcadyan-vrv9510kwac23-utils/raw/main/u-boot.ltq.lzo.nandspl. Replace the OEM bootloader with this one. Erase your NAND and write the image into it
4. Reboot the router
5. Serve the OpenWrt ramdisk image on your TFTP server and boot it via U-Boot
6. When OpenWrt boots, flash the SquashFS OpenWrt image using LuCi interface, so OpenWrt gets installed into the NAND

Signed-off-by: Daniel Huici <danielhuici@hotmail.com>

danielhuici commented:
@hauke ping

hauke commented:
Do you add support for building the bootloader to OpenWrt too?

danielhuici commented:
> Do you add support for building the bootloader to OpenWrt too?

OpenWrt U-Boot NANDSPL sources seems to be broken for OpenWrt, here's my sources repo, which is a fork of danielschwierzeck's uboot-lantiq: https://github.com/danielhuici/uboot-Arcadyan-VRV9510KWAC

Plus, I don't want to wait another half and a year to get that pushed
